algorithms/src/sorting/Kokkos_BinOpsPublicAPI.hpp

```diff
@@ -70,60 +70,65 @@ struct BinOp1D {
   }
 };
 
-template <class KeyViewType>
-struct BinOp3D {
-  int max_bins_[3] = {};
-  double mul_[3]   = {};
-  double min_[3]   = {};
+namespace Experimental {
+
+template <int DIM, class KeyViewType>
+struct BinOpND {
+  static_assert(DIM > 1);
```

Comment:
> Why can't `DIM` be 1? Is the `BinOP1D` better than this class with `DIM==1`?

I thought so too. But the interfaces are actually different. For example, the constructor takes regular integers instead of arrays, and views are 1-dimensional instead of 2-dimensional.

Reply:
We still can do `BinOP1D` with a specialization.

Reply:
Sure, we could. Would that add anything compared to the current state? I'm not sure.

Reply:
I suggest at least outputting a short message if the `static_assert` fails.
Maybe something like "Please use BinOp1D instead".

Reply:
If I put it in `Impl` namespace, it's not user accessible, so there's no reason for that.

Reply:
`BinOp1D`, `BinOp2D` and `BinOp3D` are still accessible does `BinOpND` need to be use-facing?